### ISSUE TITLE
Build time optimize HeaderSelectorProducer

### ISSUE THREAD

**Issue by davsclaus** [apache/camel-quarkus#2176]
Some Camel components uses HeaderSelectorProducer which does use reflection to discover annotations on methods on the producer class.

We can use camel quarkus to do build time optimization by scanning for those annotations (such as InvokeOnHeaders) and then pre configure this on the producer.

This may require changes in camel-core in the HeaderSelectorProducer to support such use-case.
This will also help native mode as there are no reflection and we can likely remove such rules for the extensions that uses this header selector producer.

**Comment by lburgazzoli** [comment]
@davsclaus maybe we can do some code generation (which we like a lot) on camel side so every runtime can benefit from it

**Comment by davsclaus** [comment]
Yeah we can make the camel maven plugin do that - there is about 6 or so components using this

**Comment by davsclaus** [comment]
Created a ticket
https://issues.apache.org/jira/browse/CAMEL-16102
